ci/deploy_mz_lsp_server/deploy_util.py

```diff
@@ -13,19 +13,15 @@
 import time
 from pathlib import Path
 
-try:
-    from semver.version import Version
-except ImportError:
-    from semver import VersionInfo as Version  # type: ignore
-
 import boto3
 import humanize
 
 from materialize import git
+from materialize.mz_version import MzLspServerVersion
 
 BINARIES_BUCKET = "materialize-binaries"
 TAG = os.environ["BUILDKITE_TAG"]
-VERSION = Version.parse(TAG.removeprefix("mz-lsp-server-v"))
+MZ_LSP_SERVER_VERSION = MzLspServerVersion.parse(TAG)
```

Comment:
I noticed that the **README.md** is outdated for the latest TAG format for the LSP.

The format is `BUILDKITE_TAG=mz-lsp-server-vx.y.z` rather than `BUILDKITE_TAG=x.y.z`, so this parse might fail.

Reply:
`mz-lsp-server-vx.y.z` is the correct format, right? Shall I adjust the `README.md` accordingly?
`MzLspServerVersion.parse` here is indeed incorrect, it needs to be `MzLspServerVersion.parse_mz`, which removes the prefix.

Reply:
> mz-lsp-server-vx.y.z is the correct format, right?

Correct.

> Shall I adjust the README.md accordingly?

Yes, please. I could have too, but I didn't want to mess up the PR.

Reply:
> Shall I adjust the README.md accordingly?

✅.

> `MzLspServerVersion.parse` here is indeed might be incorrect

To prevent such subtle errors, `parse()` now behaves like `parse_mz()`. To parse a version without prefix, one needs to use `parse_without_prefix()`.